service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/ReactorRunner.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.service.transport.amqps;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.apache.qpid.proton.engine.HandlerException;
+import org.apache.qpid.proton.reactor.Reactor;
+
+import java.io.IOException;
+
+@Slf4j
+public class ReactorRunner
+{
+    private final static String THREAD_NAME_PREFIX = "azure-iot-sdk-ReactorRunner-";
+    private final String threadName;
+    private final Reactor reactor;
+    public static final int REACTOR_TIMEOUT = 3141; // reactor timeout in milliseconds
+
+    public ReactorRunner(BaseHandler baseHandler, String threadNamePostfix) throws IOException
+    {
+        this.reactor = Proton.reactor(baseHandler);
+        this.threadName = THREAD_NAME_PREFIX + threadNamePostfix;
+    }
+
+    public void run(long timeoutMs)
+    {
+        Thread.currentThread().setName(this.threadName);
+
+        try
+        {
+            log.trace("Starting reactor thread {}", this.threadName);
+            this.reactor.setTimeout(REACTOR_TIMEOUT);
```

Comment:
what happens if `timeoutMs` > `REACTOR_TIMEOUT`

Reply:
I don't see this REACTOR_TIMEOUT doing anything. I'll go ahead and remove it and remove the reactor.setTimeout(...) calls